sklearn/gaussian_process/kernels.py

```diff
@@ -1141,14 +1153,23 @@ def __call__(self, X, Y=None, eval_gradient=False):
             hyperparameter of the kernel. Only returned when eval_gradient
             is True.
         """
+        length_scale = self.length_scale
+        if np.iterable(length_scale):
+            if len(length_scale) > 1:
+                length_scale = np.asarray(length_scale, dtype=np.float)
+            else:
+                length_scale = float(length_scale[0])
+        else:
+            length_scale = float(length_scale)
+
         X = np.atleast_2d(X)
-        if self.anisotropic and X.shape[1] != self.length_scale.shape[0]:
+        if self.anisotropic and X.shape[1] != length_scale.shape[0]:
             raise Exception("Anisotropic kernel must have the same number of "
```

Comment:
Not really your problem, but:

* Why is this `Exception` rather than `ValueError`???
* Elsewhere in the file is "Anisotropic kernels require that the number of length scales and features match". It should match this error message, but I also don't think that code path could ever be executed.